CONTRIBUTING.md

```diff
@@ -0,0 +1,444 @@
+
+#### orders
+
+Includes: `lims`, `status`, `osticket`
+
+The API exposes a single endpoint for submitting a batch of new samples/external samples for analysis. It handles a mix of updates to existing samples with entierly new ones. The API is designed to work well in a REST API context.
+
+The interface supports:
+
+- samples for sequencing + analysis (**scout**)
+- samples for sequencing only (**fastq**)
+- sequencing of ready-made libraries (**rml**)
+- analysis of externally sequenced samples (**external**)
+- sequencing and analysis of microbial whole genomes (**microbial**)
+
+It opens a ticket using the `osticket` API for each order which it links with the ticket number. It stores information in both LIMS and `status` for samples and pools linked by LIMS id. It stores only a minimum of information about each sample in LIMS. Most of the critial information is stored in `status` and this is also the primary place to go if we need to update e.g. application tag for a sample.
+
+#### transfer
+
+This interface has a few related roles:
+
+##### flowcell
+
+Includes: `stats`, `hk`, `status`
+
+The API accepts the name of a flowcell which will be looked up in `stats`. For all samples on the flowcell it will:
+
+1. check if the quality (Q30) is good enough to include the sequencing results.
+1. update the number of reads that the sample has been sequenced _overall_ and match this with the requirement given by the application.
+1. accordingly, the interface will look up FASTQ files and store them using `hk`.
+1. if a sample has sufficient number of reads, the `sequenced_at` date will be filled in (`status`) according to the sequencing date of the most recent flowcell.
+
+##### lims
+
+Includes: `status`, `lims`
+
+Some info if primarily stored in LIMS and needs to be syncronized over to `status`. This is the case for both the date when a samples was received and when it was finally delivered. This interface is intended to run continously as part of a crontab job.
+
+#### upload
+
+##### beacon
+
+Includes: `beacon`, `hk`, `scout`, `status`
+
+This command is used to upload variants from affected subject/s of a family to a beacon of genetic variants.
+The API will first use `status` to fetch the id of any affected subject from a given family. It will then use `hk` to retrieve a VCF file from the analyses. A temporary VCF file is then created by filtering for variants present in desired gene panel(s) (retrieved using `scout`). The `beacon` app will finally handle the upload to beacon.
+The required parameters for the upload are:
+- gene panel to use
+- id of the family
+
+##### coverage
+
+Includes: `status`, `hk`, `coverage`
+
+The API will fetch information about an analysis like name and id of the family and related samples from `status`. It will get the Sambamba output from `hk` and use the `coverage` app interface to upload the data to Chanjo for each individual sample.
```

Comment:
```suggestion
The API will fetch information about an analysis like name and family ID and related samples from `status`. It will get the Sambamba output from `hk` and use the `coverage` app interface to upload the data to Chanjo for each individual sample.
```